src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -462,11 +462,17 @@ def _compile_hermetic(self, jvm_options, ctx, classes_dir, jar_file,
       (self.post_compile_extra_resources_digest(ctx), argfile_snapshot.directory_digest)
     )
 
+    # NB: We always capture the output jar, but if classpath jars are not used, we additionally
```

Comment:
We do, because after compilation, we use a jar on the classpath 100% of the time. This is because runtime classloading (for tests) is drastically (hundreds of times) faster with jars than with files.